homeassistant/components/proximity/coordinator.py

```diff
@@ -0,0 +1,268 @@
+        # Loop through each of the distances collected and work out the
+        # closest.
+        closest_device: str | None = None
+        dist_to_zone: float | None = None
+
+        for device, zone in distances_to_zone.items():
+            if not dist_to_zone or zone < dist_to_zone:
+                closest_device = device
+                dist_to_zone = zone
+
+        # If the closest device is one of the other devices.
+        if (
+            closest_device is not None
+            and closest_device != self.state_change_data.entity_id
+        ):
+            _LOGGER.debug("closest device is one of the other devices -> unknown")
+            device_state = self.hass.states.get(closest_device)
+            assert device_state
+            return {
+                "dist_to_zone": round(distances_to_zone[closest_device]),
+                "dir_of_travel": "unknown",
+                "nearest": device_state.name,
+            }
+
+        # Stop if we cannot calculate the direction of travel (i.e. we don't
+        # have a previous state and a current LAT and LONG).
+        if (
+            self.state_change_data.old_state is None
+            or "latitude" not in self.state_change_data.old_state.attributes
+        ):
+            _LOGGER.debug("no lat and lon in old_state -> unknown")
+            return {
+                "dist_to_zone": round(
+                    distances_to_zone[self.state_change_data.entity_id]
+                ),
+                "dir_of_travel": "unknown",
+                "nearest": entity_name,
+            }
+
+        # Reset the variables
+        distance_travelled: float = 0
+
+        # Calculate the distance travelled.
+        old_distance = distance(
+            proximity_latitude,
+            proximity_longitude,
+            self.state_change_data.old_state.attributes[ATTR_LATITUDE],
+            self.state_change_data.old_state.attributes[ATTR_LONGITUDE],
+        )
+        new_distance = distance(
+            proximity_latitude,
+            proximity_longitude,
+            self.state_change_data.new_state.attributes[ATTR_LATITUDE],
+            self.state_change_data.new_state.attributes[ATTR_LONGITUDE],
+        )
+        assert new_distance is not None and old_distance is not None
+        distance_travelled = round(new_distance - old_distance, 1)
+
+        # Check for tolerance
+        if distance_travelled < self.tolerance * -1:
+            direction_of_travel = "towards"
+        elif distance_travelled > self.tolerance:
+            direction_of_travel = "away_from"
+        else:
+            direction_of_travel = "stationary"
+
+        # Update the proximity entity
+        dist_to: float | str
+        if dist_to_zone is not None:
+            dist_to = round(dist_to_zone)
+        else:
+            dist_to = DEFAULT_DIST_TO_ZONE
+
+        _LOGGER.debug(
+            "proximity.%s update entity: distance=%s: direction=%s: device=%s",
+            self.friendly_name,
```

Comment:
Why do we use something that wd call the coordinator's friendly name to create what looks like and entity_id?

If we want to log an entity_id, why not log the entity_id of the entity we're interested in?

Reply:
5bca7f49c80fa615f8e7a1342abcdf6554fd97c5